sdk/playwrighttesting/microsoft-playwright-testing/src/utils/reporterUtils.ts

```diff
@@ -317,20 +319,38 @@ class ReporterUtils {
     },
   };
 
-  public static isTimeGreaterThanCurrentPlus10Minutes(isoString: string): boolean {
-    // Parse ISO 8601 string into a timestamp
-    const timestampFromIsoString: number = Date.parse(isoString);
-    // Calculate the current timestamp plus 10 minutes
-    const currentTimestampPlus10Minutes: number = Date.now() + 10 * 60 * 1000;
-    // Compare the timestamps
-    return timestampFromIsoString > currentTimestampPlus10Minutes;
+  // eslint-disable-next-line @azure/azure-sdk/ts-use-interface-parameters
+  public static isTimeGreaterThanCurrentPlus10Minutes(sasUri: StorageUri): boolean {
+    try {
+      const url = new URL(sasUri.uri);
+      const params = new URLSearchParams(url.search);
+      const expiryTime = params.get("se"); // 'se' is the query parameter for the expiry time
+      if (expiryTime) {
+        const timestampFromIsoString = new Date(expiryTime).getTime();
+        const currentTimestampPlus10Minutes = Date.now() + 10 * 60 * 1000;
+        const isSasValidityGreaterThanCurrentTimePlus10Minutes =
+          timestampFromIsoString > currentTimestampPlus10Minutes;
+        if (!isSasValidityGreaterThanCurrentTimePlus10Minutes) {
+          reporterLogger.info(
+            `Sas rotation required because close to expiry, SasUriValidTillTime: ${timestampFromIsoString}, CurrentTime: ${currentTimestampPlus10Minutes}`,
+          );
+        }
+        return isSasValidityGreaterThanCurrentTimePlus10Minutes;
+      }
+      reporterLogger.error(`Sas rotation required because expiry param not found.`);
```

Comment:
will it show up without DEBUG variable ?

Reply:
No it won't I tested it.